### PR TITLE
Optimize db benchmarks for Actix framework

### PR DESCRIPTION
**Pull request by fafhrd91**

### CODE REVIEW

File: frameworks/Rust/actix/Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ num_cpus = "1.0"
 futures = "0.1"
 http = "0.1"
 actix = "^0.4.3"
-actix-web = "=0.3.1"
+actix-web = "0.3"
```
Comment (nbrady-techempower) [inline frameworks/Rust/actix/Cargo.toml]: Just curious, but why are you regressing here?

Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.toml]: it just "any available version 0.3.x", it it will use current version 0.3.3

Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.toml]: here is link to travis job, https://travis-ci.org/TechEmpower/FrameworkBenchmarks/jobs/338621337#L2183

Comment (nbrady-techempower) [reply frameworks/Rust/actix/Cargo.toml]: What's the difference between "^0.4" and "0.4" is that still getting you "0.4.*"? Where possible, we'd like to tie down packages, even to their patch version. The reason for this is, when we run the benchmark rounds and tag the commit, we want to be able to replicate the runs at any time. "0.3" could be "0.3.22" in some future replication, and could create radically different results (or even break the tests - sadly non-breaking semantic versioning is a utopia I only dream of).

Comment (nbrady-techempower) [reply frameworks/Rust/actix/Cargo.toml]: In fact, we should probably be adding the `cargo.lock` for all rust frameworks

Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.toml]: "^" mean lowest allowed version. i.e I can say "^0.3.2" - any version starting "0.3.2", "0.3" is any "0.3.x" version. to pin specific version use "=0.3.2" - only 0.3.2. 

Cargo.lock file contains specific version, it solves reproducible builds problem

Comment (nbrady-techempower) [reply frameworks/Rust/actix/Cargo.toml]: Fantastic, as we just had this issue with a few rust frameworks breaking randomly, let's go with a cargo.lock here.

Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.toml]: do you want me to commit Cargo.lock for actix in this PR?

Comment (nbrady-techempower) [reply frameworks/Rust/actix/Cargo.toml]: That would be great @fafhrd91 !

Comment (fafhrd91) [reply frameworks/Rust/actix/Cargo.toml]: added

**Comment by fafhrd91** [comment]
could you guys merge this before next test run? 

btw are you planing to do another test run?

**Comment by nbrady-techempower** [comment]
We've already captured R15 results and are preparing things for release, so this will not make it. However, you can see how it performs here: https://tfb-status.techempower.com/

**Comment by fafhrd91** [comment]
ok, thanks

File: frameworks/Rust/actix/src/main.rs [hunk 2]
```diff
@@ -48,32 +47,26 @@ fn world_row(req: HttpRequest<State>) -> Box<Future<Item=HttpResponse, Error=Err
 fn queries(req: HttpRequest<State>) -> Box<Future<Item=HttpResponse, Error=Error>> {
     // get queries parameter
     let q = if let Some(q) = req.query().get("q") {
-        q.parse::<usize>().ok().unwrap_or(1)
+        q.parse::<u16>().ok().unwrap_or(1)
     } else {
         1
     };
```
Comment (nbrady-techempower) [inline frameworks/Rust/actix/src/main.rs]: I've been playing around with rust a bit. These 5 lines could be:

`let q = req.query().get("q").unwrap_or("1").parse::<u16>().unwrap_or(1);`

Happens in updates too in this file, which is still using `<usize>`

Not a nit or anything, just enjoying messing around with the language.

Comment (fafhrd91) [reply frameworks/Rust/actix/src/main.rs]: well, if this ```.unwrap_or("1")``` happen, then you would parse default string.

equivalent would be:
```rust
let q = req.query().get("q").unwrap_or(1).map(|v| v.parse::<u16>().unwrap_or(1));
```

Comment (fafhrd91) [reply frameworks/Rust/actix/src/main.rs]:
```rust
let q: u16 = req.query().get("q").map(|v| v.parse().unwrap_or(1)).unwrap_or(1);
```

**Pull request merged**